test/interop/main_client.test.ts

```diff
@@ -0,0 +1,63 @@
+import { RemoteStreamRegistry, StreamClient} from "../../src";
+import { firstValueFrom, map, take, toArray } from "rxjs";
+import { Offset } from "../../src/gen/model/v1/model";
+import { strict as assert } from "assert";
+
+//const testEndpoint = "https://stream-api.cluster.amur-dev.proxima.one";
+const testEndpoint = "http://localhost:7000"
+jest.setTimeout(30000)
+
+describe("StreamClient", () => {
+  let client: StreamClient;
+  beforeEach(() => {
+    client = new StreamClient({registry: new RemoteStreamRegistry(testEndpoint)});
+  });
+
+  it("should get stream endpoints and data", async () => {
+    jest.setTimeout(30000)
+    const name = "eth-main-blockheader0.new-runtime"
+    const stream =  await client.getStream(name)
+    expect(stream.metadata).toMatchSnapshot();
+    expect(stream.endpoints).toBeDefined()
+    expect(stream.name).toMatch(name)
+  });
+
+  it("should get an offset for a specified height and/or timestamp", async () => {
+    jest.setTimeout(30000)
+    const name = "eth-main-blockheader0.new-runtime"
+    const stream =  await client.getStream(name)
+    expect(stream).toBeDefined()
+    expect(stream.stats).toBeDefined()
+    assert(stream.stats)
+    const height = stream.stats.start.height || 0
+    const offset = await client.findOffset(name, {height: Number(height)})
+    expect(offset).toBeDefined()
+    expect(offset).toMatchSnapshot()
```

Comment:
:thinking: Didn't know about it. But I think the snapshot file has to also be commited for these tests to work.

Reply:
True. Done